Model/src/BoardPackage/JailSquare.java

```diff
@@ -1,7 +1,83 @@
 package BoardPackage;
 
-/**
- * Created by Dylan Kernohan on 2/10/18.
- */
-public class JailSquare {
+import GamePackage.Player;
+
+import java.util.ArrayList;
+
+/*********************************************************************
+ * The JailSquare Class tracks what players are in jail.
+ *
+ * @author David Baas
+ * @version 2/14/2018
+ ********************************************************************/
+public class JailSquare extends BoardSquare {
+
+    /**An ArrayList to store what players are in jail.*/
+    private ArrayList<Player> inJail;
+
+    /*****************************************************************
+     * The default constructor creates a JailSquare setting its name
+     * variable and initializes the inJail ArrayList.
+     *
+     * @param name sets the name of the board square.
+     ****************************************************************/
+    public JailSquare(String name){
+
+        super(name);
+
+        inJail = new ArrayList<Player>();
+    }
+
+    /*****************************************************************
+     * The addToJail method adds a single player into the jail list
+     * There is a check when this method is invoked that the same
+     * player is not added to jail twice.
+     *
+     * @param p the Player who is being sent to jail.
+     ****************************************************************/
+    public void addToJail(Player p){
+        for(Player o: inJail){

```

Comment:
As a small comment on the side I wanted to point out that this method might make it hard to debug or find errors. It might make it easier to get rid of this logic and let the game check for the posible error. Or we can have this method throw and error, but I think that wouldn't be the best plan of action.

Reply:
In a lot of other places we take checks out and will do them in the Game class. I agree Santiago